camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusHourlySweeper.java

```diff
@@ -0,0 +1,327 @@
+      String key = (String) pair.getKey();
+      conf.set(key, (String) pair.getValue());
+    }
+    for (Properties jobProps : planner.getOutlierProperties()) {
+      this.executorService.submit(new OutlierCollectorRunner(jobProps, FileSystem.get(conf)));
+    }
+  }
+
+  private void reportMetrics() throws IllegalArgumentException, IOException {
+    LOG.info("reporting metrics");
+    metrics.reportTotalRunningTime();
+    metrics.reportTotalDataSize();
+    metrics.reportDataSizeByTopic();
+    metrics.reportDurationFromStartToRunnerStart();
+    metrics.reportDurationFromRunnerStartToMRSubmitted();
+    metrics.reportDurationFromMRSubmittedToMRStarted();
+    metrics.reportDurationFromMRStartedToMRFinished();
+    LOG.info("finished reporting metrics");
+  }
+
+  public static void main(String args[]) throws Exception {
+    CamusSweeper job = new CamusHourlySweeper();
+    ToolRunner.run(job, args);
+  }
+
+  @Override
+  protected Future<?> runCollector(Properties props, String topic) {
+    String jobName = topic + "-" + UUID.randomUUID().toString();
+    props.put("tmp.path", props.getProperty(CAMUS_SWEEPER_TMP_DIR) + "/" + jobName + "_" + System.currentTimeMillis());
+
+    if (props.containsKey(REDUCE_COUNT_OVERRIDE + topic))
+      props.put("reducer.count", Integer.parseInt(props.getProperty(REDUCE_COUNT_OVERRIDE + topic)));
+
+    LOG.info("Processing " + props.get(INPUT_PATHS));
+
+    return executorService.submit(new KafkaCollectorRunner(jobName, props, errorMessages, topic));
+  }
+
+  @Override
+  protected void runCollectorForTopicDir(FileSystem fs, String topic, Path topicSourceDir, Path topicDestDir)
+      throws Exception {
+    LOG.info("Running collector for topic " + topic + " source:" + topicSourceDir + " dest:" + topicDestDir);
+    List<Future<?>> tasksToComplete = new ArrayList<Future<?>>();
+
+    List<Properties> jobPropsList = planner.createSweeperJobProps(topic, topicSourceDir, topicDestDir, fs, metrics);
+
+    for (Properties jobProps : jobPropsList) {
+      tasksToComplete.add(runCollector(jobProps, topic));
+    }
+  }
+
+  private class KafkaCollectorRunner extends CamusSweeper.KafkaCollectorRunner {
+
+    public KafkaCollectorRunner(String name, Properties props, List<SweeperError> errorQueue, String topic) {
+      super(name, props, errorQueue, topic);
+    }
+
+    @Override
+    public void run() {
+      KafkaCollector collector = null;
+      try {
+        LOG.info("Starting runner for " + this.props.getProperty(TOPIC_AND_HOUR));
+        collector = new KafkaCollector(props, name, topic);
+        LOG.info("Running " + name + " for input " + props.getProperty(INPUT_PATHS));
+        collector.run();
+      } catch (Throwable e) { // Sometimes the error is the Throwable, e.g. java.lang.NoClassDefFoundError
+        e.printStackTrace();
+        LOG.error("Failed for " + name + " ,job: " + collector == null ? null : collector.getJob() + " failed for "
```

Comment:
Seems better to use String.format() here. Also you can do `LOG.error(msg, e);` so you don't need to include the exception message in `msg`.